templates/undercloud-networks.yaml

```diff
@@ -62,6 +67,31 @@ resources:
       gateway_ip: null
       enable_dhcp: false
 
+  public_router:
+    type: OS::Neutron::Router
+    properties:
+      name: public-router
+      external_gateway_info:
+        network: {get_param: external_net}
+
+  public_router_port:
+    type: OS::Neutron::Port
+    properties:
+      network: {get_resource: public_network}
+      port_security_enabled: false
+      fixed_ips:
+       - ip_address: {get_param: public_net_router_address}
+
+  public_router_interface:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router: {get_resource: public_router}
+      port: {get_resource: public_router_port}
+
+  external_net:
```

Comment:
This needs to go in parameters, not resources.

Reply:
*facepalm*